javaparser-core/src/main/java/com/github/javaparser/JavaToken.java

```diff
@@ -159,5 +155,19 @@ public JavaToken orIfInvalid(JavaToken anotherToken) {
         return anotherToken;
     }
 
+    public enum Category {
+        WHITESPACE, COMMENT, IDENTIFIER, KEYWORD, LITERAL, SEPARATOR, OPERATOR
+    }
+
+    public JavaToken.Category getCategory() {
+        return TokenTypes.getCategory(kind);
+    }
 
-}
+    public boolean isEndOfLineToken() {
```

Comment:
According to the JLS it's part of whitespace (although it's not even a token according to the JLS) - I was thinking about subcategories but there is not much to subcategorize. Maybe have a `JlsCategory` and a `FineGrainedCategory` ?